src/classes/BDI_AdditionalObjectService.cls

```diff
@@ -279,7 +291,90 @@ public with sharing class BDI_AdditionalObjectService {
 
 
     /*******************************************************************************************************
-    * @description Determines if the predecessor to this Data Import Object Mapping exists
+    * @description For newly created parent sobjects, this method updates the appropriate child predecessor
+    * record's lookup field to the id of the newly created object.
+    * @param objWrapsForChildPredecessorUpdate list of object wraps that were successfully created
+    * and need to update their child predecessors.
+    */
+    private void updateChildPredecessorRecords(BDI_ObjectWrapper[] objWrapsForChildPredecessorUpdate){
+
+        Map<String,Map<Id,SObject>> childSObjectMapByObjectAPIName = new Map<String,Map<Id,SObject>>();
+        Map<Id,BDI_ObjectWrapper[]> successorObjWrapsByPredecessorId = new Map<Id,BDI_ObjectWrapper[]>();
+
+        for (BDI_ObjectWrapper objWrap : objWrapsForChildPredecessorUpdate) {
+
+            Id predecessorId = (Id)objWrap.dataImport.get(objWrap.predecessorObjMapping.Imported_Record_Field_Name__c);
+            SObject sObjForUpdate;
+
+            Map<Id,SObject> tempSObjMap = new Map<Id,SObject>();
+
+            if (childSObjectMapByObjectAPIName.get(objWrap.predecessorObjMapping.Object_API_Name__c) != null) {
+                tempSObjMap = childSObjectMapByObjectAPIName.get(objWrap.predecessorObjMapping.Object_API_Name__c);
+            }
+
+            // Attempt to find if the predecessor sobject if it is already in the map for update and update 
+            // the existing SObject if possible.  Otherwise initialize a new SObject for update.
+            if (tempSObjMap.size() > 0 && tempSObjMap.get(predecessorId) != null) {
+                sObjForUpdate = tempSObjMap.get(predecessorId);
+            } else {
+                sObjForUpdate = UTIL_Describe.getPrototypeObject(objWrap.predecessorObjMapping.Object_API_Name__c);
+            }
+
+            // Set the id of the sobject to the predecessors record id
+            sObjForUpdate.put('Id',predecessorId);
+
+            // Place the Id of the newly created SObject into the appropriate relationship field on the child.
+            sObjForUpdate.put(objWrap.objMapping.Relationship_Field__c,objWrap.sObj.Id);
+
+            // Put the SObject back into the enclosing maps.
+            tempSObjMap.put(predecessorId, sObjForUpdate);
+            childSObjectMapByObjectAPIName.put(objWrap.predecessorObjMapping.Object_API_Name__c,tempSObjMap);
+
+            // Store the successor object wraps by their predecessor id for later use in recording errors.
+            if (successorObjWrapsByPredecessorId.get(predecessorId) != null) {
+                successorObjWrapsByPredecessorId.get(predecessorId).add(objWrap);
+            } else {
+                successorObjWrapsByPredecessorId.put(predecessorId, new BDI_ObjectWrapper[]{objWrap});
+            }
+        }
+
+        if (childSObjectMapByObjectAPIName.size() > 0) {
```

Comment:
I think we can skip this "if" statement - if the childSObjectMapByObjectAPIName is empty, the for loop should just never iterate on the empty keyset, right?

Reply:
removed